skiko/src/iosMain/kotlin/org/jetbrains/skiko/context/MetalContextHandler.kt

```diff
@@ -69,6 +68,8 @@ internal class MetalContextHandler(layer: SkiaLayer) : ContextHandler(layer, lay
         super.flush()
         surface?.flushAndSubmit()
         metalRedrawer.finishFrame()
+
+        disposeCanvas()
```

Comment:
Why this move?

Reply:
Because canvas (wrapping MTLDrawable owned texture) should be disposed in the end of the frame, not in the beginning of a new one on Metal.